includes/civicrm.shortcodes.php

```diff
@@ -704,7 +704,20 @@ public function preprocess_atts($atts) {
         if ($mode == 'preview' || $mode == 'test') {
           $args['action'] = 'preview';
         }
-        $args['q'] = 'civicrm/pcp/info';
+
+        switch ($action) {
+          case 'transact':
+            $args['q'] = 'civicrm/contribute/transact';
+            $args['pcpId'] = $args['id'];
+            break;
+
+          case 'info':
+            $args['q'] = 'civicrm/pcp/info';
+            break;
+
+          default:
+            return FALSE;
```

Comment:
@mattwire Shouldn't the default be the same as `case 'info':` to preserve existing behaviour?

Reply:
@christianwach Yes agreed. I've left the case 'info': in and a fall-through to default. I know that's not needed but it seems clearer to someone reading the code.

Reply:
@mattwire Agreed, looks good now.